### ISSUE TITLE
Rule Change: add "allowProperties" option to `no-restricted-properties`

### ISSUE THREAD

**Issue by erosman** [eslint/eslint#19762]
### What rule do you want to change?

no-restricted-properties

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/* global obj */

obj.a = 1;
obj.b = 2;
obj.c = 3;

obj.x = 4;
obj.y = 4;
```

### What does the rule currently do for this code?

As per the issue mentioned in [Define specific object properties in globals](https://github.com/eslint/eslint/discussions/19758), additional "allowProperties" option in `no-restricted-properties` could be beneficial and avoids having to generate a large rule list by entering them individually.

```json
{
  "rules": {
    "no-restricted-properties": [2, {
        "object": "obj",
        "allowProperties": [ "a", "b", "c" ],
        "message": "Property is not supported."
    }]
  }
}
```

### What will the rule do after it's changed?

Generates error for `obj.x`, `obj.y`,  etc
```
'obj.x' is restricted from being used. Property is not supported. (no-restricted-properties) ESLint
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Makes sense to me 👍 

@eslint/eslint-team thoughts?

**Comment by Tanujkanti4441** [comment]
> @Tanujkanti4441 I actually started work on this 😅.

That's great! Let's wait for the acceptance of the issue then.

**Comment by nzakas** [comment]
Three team members gave :+1: so marking as accepted. 

@erosman we cannot guarantee when a feature will land. It depends on team members' availability and approval.
